File: terre05.exs
# Division

# Helper function
parse_to_int = fn str ->
  case Integer.parse(str) do
    {number, _reste} -> {:ok, number}
    :error -> {:error, str}
  end
end

System.argv()
|> case do
  [arg1, arg2 | _rest] ->
    with {:ok, number1} <- parse_to_int.(arg1),
         {:ok, number2} <- parse_to_int.(arg2) do
      case {number1, number2} do
        {_n1, 0} ->
          IO.puts("Cannot divide by 0")

        {n1, n2} when n1 < n2 ->
          IO.puts("Second argument must be lower than the first")

        {n1, n2} ->
          IO.puts("résultat: #{floor(n1 / n2)}")
          IO.puts("résultat: #{rem(n1, n2)}")
      end
    else
      {:error, invalid} -> IO.puts("Expected a valid number got=#{invalid}")
    end

  _args ->
    IO.puts("Usage: elixir terre05.exs <number1> <number2>")
end

File: terre05.js
// Divisions
const args = process.argv.slice(2)
if (args.length != 2) {
    console.log("Usage: node terre05.js <number1> <number2>")
    process.exit(1)
}

const arg1 = args[0]
const arg2 = args[1]
const number1 = parseInt(arg1)
const number2 = parseInt(arg2)

if (isNaN(number1)) {
    console.log(`Expected a valid number got=${arg1}`)
    process.exit(1)
}

if (isNaN(number2)) {
    console.log(`Expected a valid number got=${arg2}`)
    process.exit(1)
}

if (number2 == 0) {
    console.log("Cannot divide by 0")
    process.exit(1)
}

if (number1 < number2) {
    console.log("Second argument must be lower than the first")
    process.exit(1)
}

console.log(`résultat: ${Math.floor(number1 / number2)}`)
console.log(`reste: ${Math.round(number1 % number2)}`)
